grid-proxy/internal/explorer/server.go

```diff
@@ -467,6 +439,61 @@ func (a *App) getNodeGpus(r *http.Request) (interface{}, mw.Response) {
 	return res, mw.Ok()
 }
 
+// getContract godoc
+// @Summary Show single contract info
+// @Description Get data about a single contract with its id
+// @Tags Contract
+// @Param contract_id path int yes "Contract ID"
+// @Accept  json
+// @Produce  json
+// @Success 200 {object} types.Contract
+// @Failure 400 {object} string
+// @Failure 404 {object} string
+// @Failure 500 {object} string
+// @Router /contracts/{contract_id} [get]
+func (a *App) getContract(r *http.Request) (interface{}, mw.Response) {
+	contractID := mux.Vars(r)["contract_id"]
```

Comment:
what if contract_id is a invalid number, shouldn't we return  a bad request errror?
if so, i believe this contract id conversion should happen here, no?

Reply:
yes indeed. the conversions happen first thing in the next called method `a.getContractData(contractID)` just tended to do the same as in `getNode` function :D